aws/resource_aws_backup_vault_test.go

```diff
@@ -0,0 +1,104 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/backup"
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAwsBackupVault_basic(t *testing.T) {
+	rInt := acctest.RandInt()
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsBackupVaultDestroy,
```

Comment:
I'm not sure if this is an issue with AWS or not, but I get a weird permission issue when I hit `CheckDestroy`:
```
$ make testacc TESTARGS='-run=TestAccAwsBackupVault_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v -parallel 20 -run=TestAccAwsBackupVault_basic -timeout 120m
?   	github.com/terraform-providers/terraform-provider-aws	[no test files]
=== RUN   TestAccAwsBackupVault_basic
=== PAUSE TestAccAwsBackupVault_basic
=== CONT  TestAccAwsBackupVault_basic
--- FAIL: TestAccAwsBackupVault_basic (7.21s)
    testing.go:599: Error destroying resource! WARNING: Dangling resources
        may exist. The full state and error is shown below.
        
        Error: Check failed: AccessDeniedException: Insufficient privileges to perform this action.
        	status code: 403, request id: 1cea279f-dca0-405a-98e6-a2d112b9d9de
        
        State: <no state>
FAIL
FAIL	github.com/terraform-providers/terraform-provider-aws/aws	7.232s
GNUmakefile:20: recipe for target 'testacc' failed
make: *** [testacc] Error 1
```

Reply:
When I remove the `CheckDestroy` my test passes successfully.  Weird, because I'm just doing the same type of request here that I'm doing with the `Read` function.

Reply:
User Error.  Also, I wasn't handling the err logic correctly.